test/jdk/java/awt/PopupMenu/PopupMenuStayOpen.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4791953
+  @requires (os.family == "linux" | os.family == "mac")
+  @summary Checks that popup menu stay open after a triggering click.
+  @key headful
+  @run main/othervm -Dsun.java2d.uiScale=1 PopupMenuStayOpen
```

Comment:
Why do you need to set uiscale=1 in this test?

Reply:
If the uiscale is not set then the position of popup menu varies according to display screen, like may not be always from bottom-left of cursor (when mouse click happens). Since we are only open sourcing the test, so had to set the scale to 1 so that it passes in all conditions.